tpot/gp_deap.py

```diff
@@ -75,32 +104,22 @@ def varOr(population, toolbox, lambda_, cxpb, mutpb):
     1 - *cxpb* - *mutpb*.
     """
     offspring = []
+    
     for _ in range(lambda_):
         op_choice = np.random.random()
         if op_choice < cxpb:  # Apply crossover
-            idxs = np.random.randint(0, len(population), size=2)
-            ind1, ind2 = toolbox.clone(population[idxs[0]]), toolbox.clone(population[idxs[1]])
-            ind_str = str(ind1)
-            num_loop = 0
-            while ind_str == str(ind1) and num_loop < MAX_MUT_LOOPS:
-                ind1, ind2 = toolbox.mate(ind1, ind2)
-                num_loop += 1
-            if ind_str != str(ind1):  # check if crossover happened
-                del ind1.fitness.values
+            ind1, ind2 = pick_two_individuals_eligible_for_crossover(population)  
+            ind1, _ = toolbox.mate(ind1, ind2)  
+            del ind1.fitness.values
             offspring.append(ind1)
         elif op_choice < cxpb + mutpb:  # Apply mutation
-            idx = np.random.randint(0, len(population))
-            ind = toolbox.clone(population[idx])
-            ind_str = str(ind)
-            num_loop = 0
-            while ind_str == str(ind) and num_loop < MAX_MUT_LOOPS:
-                ind, = toolbox.mutate(ind)
-                num_loop += 1
-            if ind_str != str(ind):  # check if mutation happened
-                del ind.fitness.values
+            idx = np.random.choice(len(population))
+            ind = population[idx]
+            ind, = toolbox.mutate(ind)            
+            del ind.fitness.values
             offspring.append(ind)
         else:  # Apply reproduction
-            idx = np.random.randint(0, len(population))
+            idx = np.random.choice(len(population))
```

Comment:
It might be preferable to use `randint` here because it's more explicit. It wasn't immediately obvious that `choice` was correct here until I looked at the docs to confirm that `np.random.randint(0, len(population))` == `np.random.choice(len(population))`.